demos/notebooks/demo_pipeline.ipynb

```diff
@@ -610,10 +1023,736 @@
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<a id=display_large></a>\n",
+    "## Appendix 3: Displaying large files\n",
+    "The way we have loaded movie objects in the main demo requires loading all of the data you want to view into memory, and this is not going to be possible with extremely large data sets. But even with very large data sets, you typically want to visualize what is going on, make sure things seem reasonable, etc. Caiman has built-in tools to handle this case. One, you just load *some* of a movie into a movie object using the `subindices` argument to the `load()` function. For example, if you just want to load the first 500 frames of a movie, you can send it `subindices=np.arange(0,500)`. \n",
+    "\n",
+    "Probably even better, there is a `play_movie()` function that behaves just like `movie.play()` in the main demo, that doesn't ever load the movie into memory, but just takes the filename as an argument and iteratively loads frames from disk and shows them as needed. If you want to play with it, just import with `caiman.base.movies import play_movie` and read the documentation. We don't use it for the demo because the demo movie is small, and we actually do some calculations on the loaded movie array in the demo. This may change in the future. \n",
+    "\n",
+    "Another option for viewing very large movies is to use the [fastplotlib](https://github.com/fastplotlib/fastplotlib) library, which leverages GPU-based visualization. Like `play_movie()`, it uses lazy loading so will not overload your memory, and provides interactive Jupyter widgets to explore your movies. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<a id=caiman_estimators></a>\n",
+    "## Appendix 4: Estimators and parameters\n",
+    "\n",
+    "For the main computations in the pipeline -- like motion correction and CNMF -- Caiman breaks things into two steps:\n",
+    "\n",
+    "- Initialize the estimator object (e.g., `MotionCorrect`, `CNMF`) by sending it the set of parameters it will use. \n",
+    "- Run the method on the object to generate the results. For `CNMF` this will be the `fit()` method, for motion correction it is `motion_correct()`.\n",
+    "\n",
+    "This modular architecture, where models are initialized with parameters, and then estimates are made with a separate call to a method that carries out the calculations on data fed to the model, is useful for a few reasons. The main reason is that it allows for efficient exploration of parameter space. Often, after setting some *initial* set of parameters, you will want to modify the parameters after visualizing your data (e.g., after viewing the size of the neurons). \n",
+    "\n",
+    "Note that our API is like that used by the [scikit-learn](https://scikit-learn.org/stable/) machine learning library. From their [manuscript on api design](https://arxiv.org/abs/1309.0238):\n",
```

Comment:
Is this important to talk about?

Reply:
For non-programmers it is confusing to have this modularity, and I get questions about this at workshops sometimes -- people get confused about why we don't just initialize and fit all at once.  It intuitively seems inefficient to have it broken up.

Reply:
Moved to docs